README.md

```diff
@@ -8,19 +8,50 @@
 
 # Installation 
 
-1. `npm install screepsmod-auth` in your server folder.
-2. Thats it!
+1. Either run `npm install screepsmod-auth` in your screeps private server folder or subscribe the Steam workshop  https://steamcommunity.com/sharedfiles/filedetails/?id=800390576 if your using the in-game private server.
 
-# Usage
+2. Modify mods.json and add a new line in your mods list pointing to this mod like so:
 
-## Steam Console method
-1. Open the steam screeps client
-2. In console, run `setPassword('YourDesiredPassword')`
-3. Now you should be able to login via API
+```
+  "mods": [
+    "node_modules\\screepsmod-auth\\index.js"
+  ],
+```
+3. Relaunch your screeps private server.
+3. Follow the 
+[CLI steps](#screeps-server-command-line-interface-cli-method) below to setup your username and password 
+
+# Setup
+
+## Screeps Server Command Line Interface (CLI) method
+1. Connect to your screeps private server once to create your account.  You don't need to set an email as that won't be used.  Just username and password is important.
+
+2. Run the server's command line interface (CLI) either in Steam, or if your running a standalone server as described here https://github.com/screeps/screeps#command-line-interface-cli
+
+3. In server's CLI, run 
+
+```
+setPassword('yourUserName', 'yourPassword')
+```
+Note: that setPassword says to pass in the email.  Don't do this.  Instead give it your username.
+
+4. If you get back an error that setPassword isn't defined, then its likely your mod isn't installed.  See the mod.json set above to fix it.  If setPassword returns {} then that's also a failure as your user name wasn't found.  If setPassword returns { modified: 1 }, then it succceded.
+
+5. Now you should be able to login 
 
 ## Web Form Method
-1. Open the steam client at least once (Required to create initial account)
-2. Goto http://yourServerHostOrIP:21025/authmod/password/
+1. Connect to your screeps private server once to create your account.  You don't need to set an email as that won't be used.  Just username and password is important.
+
+2. Send a POST call to http://yourServerHostOrIP:21025/authmod/password/ with this header:
```

Comment:
This will not work, it requires a valid login token in order to set the password. The web form takes care of that by having the steam login button to obtain the token.

Reply:
So does the web form method only support steam then?  I'm using the npm method and I had to use fiddler, edit the POST call according and reply it, and it returned 200.   Not sure if it actually worked however

Reply:
It support both steam and npm, it uses the steam oauth system to get a valid steam auth, which the server then returns a server auth token to the form. It uses that token for authentication to set the password. it needs to authenticate the user first in order to prevent just anyone from setting the password. Since it uses the steam login system, it also auto-creates the user if it doesn't exist. However, I have been seeing it take two tries to set the password. I haven't had the time to poke at that and figure out the cause yet.